### ISSUE TITLE
taxon legend from map whis atlas

### ISSUE THREAD

**Issue by Geoscope19** [qgis/QGIS#54654]
### What is the bug or the crash?

Hello,

I make an atlas based on the layer "Périmètre de l'OAP"
- objects in this layer can be close to each other so that on a map of "polygon 1", part of the "polygon 2" can appear
- the layer "Périmètre de l'OAP" is configured in the atlas so that only the one concerned by the atlas entity appears: it works well (black arrow from map - "cartes 1, 2 and 3" in exemple : [https://geoscope.fr/nextcloud/index.php/s/t6knHdtDHC6EgfE](url)
- the layer "Enjeux OAP surf UD" is configured so that they appear completely even if none of their polygon intersects the targeted polygon. Objects appear logically in legend (blue arrow) it works well
- layer "Enjeux OAP surf VE" should only appear in the targeted perimeter: it works BUT in legend appears the taxa "verger" and "espace vert urbain" which are hidden because only concerns the perimeter of the neighboring object. The objects are well hidden, but they appear in legend, which I do not want (red arrow).
The two taxa "verger" and "espace vert urbain" are not visible, they should not appear in legend. Is this a bug?

### Steps to reproduce the issue

Screen capture for exemple : [https://geoscope.fr/nextcloud/index.php/s/t6knHdtDHC6EgfE](url)

carte 1 : map canvas whis "perimetre 1" and "perimetre 2"
carte 2 : relation between object on map and taxon of legend
carte 3 : detail of the configuration of layers in the atlas

### Versions

<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

Version de QGIS | 3.32.2-Lima | Révision du code | c0b8833964
-- | -- | -- | --
Version de Qt | 5.15.3
Version de Python | 3.9.5
Version de GDAL/OGR | 3.7.1
Version de Proj | 9.2.1
Version de la base de données du registre EPSG | v10.088 (2023-05-13)
Version de GEOS | 3.12.0-CAPI-1.18.0
Version de SQLite | 3.41.1
Version de PDAL | 2.5.5
Version du client PostgreSQL | 15.2
Version de SpatiaLite | 5.1.0
Version de QWT | 6.1.6
Version de QScintilla2 | 2.13.1
Version de l'OS | Windows 10 Version 2009

</body></html>Version de QGIS
3.32.2-Lima
Révision du code
[c0b8833964](https://github.com/qgis/QGIS/commit/c0b8833964)
Version de Qt
5.15.3
Version de Python
3.9.5
Version de GDAL/OGR
3.7.1
Version de Proj
9.2.1
Version de la base de données du registre EPSG
v10.088 (2023-05-13)
Version de GEOS
3.12.0-CAPI-1.18.0
Version de SQLite
3.41.1
Version de PDAL
2.5.5
Version du client PostgreSQL
15.2
Version de SpatiaLite
5.1.0
Version de QWT
6.1.6
Version de QScintilla2
2.13.1
Version de l'OS
Windows 10 Version 2009

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by DelazJ** [comment]
Having helped @Geoscope19  to debug QGIS behavior here, I would add that the main point is: if you use the "Layout map item properties --> Clipping settings --> Clip to atlas feature" option, features that are no longer visible in the map extent are still reported in the legend, regardless the use of "Only show items inside (linked maps | current atlas feature)" checkbox.
Looks like legend widget does not take the clipping options into consideration.

**Comment by elpaso** [comment]
@DelazJ a minimal sample project to reproduce the issue could be helpful.

**Comment by Geoscope19** [comment]
My sample from only 3 layers
[OAP.zip](https://github.com/qgis/QGIS/files/12684182/OAP.zip)

**Comment by Geoscope19** [comment]
and file  projet... sorry !
[sample light.zip](https://github.com/qgis/QGIS/files/12684220/sample.light.zip)

**Comment by elpaso** [comment]
@Geoscope19 I am testing your sample project but I cannot reproduce the issue, is there any specific atlas feature ID that I should look at?

It seems to me that the legend works as expected and it does not show any Vegetalisation item which is outside the atlas feature.

**Comment by elpaso** [comment]
@Geoscope19 nevermind: La Croix Saint-Martin seems to show the bug

**Comment by Geoscope19** [comment]
Yes. and : La moissie, la clauzel, le pradal, les plaines, luziers nord, Pech-Bracou

**Comment by Geoscope19** [comment]
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN" "http://www.w3.org/TR/REC-html40/strict.dtd">
<html><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><style type="text/css">
p, li { white-space: pre-wrap; }
</style></head><body>

Version de QGIS | 3.36.0-Maidenhead | Révision du code | 09951dc0
-- | -- | -- | --

Hello, this version of Qgis reproduces this bug

</body></html>Version de QGIS
3.36.0-Maidenhead
Révision du code
[09951dc0](https://github.com/qgis/QGIS/commit/09951dc0)
